File: JAVA/DPJ/src/main/java/abstract_factory/ConcreteLogFileProduct.java
package abstract_factory;

import java.io.FileWriter;
import java.io.IOException;

public class ConcreteLogFileProduct implements AbstractLogProduct{
    @Override
    public boolean log(String message) {
        try (var f = new FileWriter("log.txt")) {
            f.write(message);
        } catch (IOException e) {
            return false;
        }
        return true;
    }
}

File: KOTLIN/DPK/src/main/kotlin/creational/ConcreteLogFileProduct.kt
package creational

import java.io.FileWriter
import java.io.IOException

class ConcreteLogFileProduct : AbstractLogProduct {
    override fun log(message: String): Boolean {
        try {
            val f = FileWriter("log.txt")
            f.write(message)
        } catch (e: IOException) {
            return false
        }

        return true
    }

}
